os/net/lwip/src/api/sockets.c

```diff
@@ -393,7 +393,12 @@ struct lwip_sock *get_socket(int s, pid_t pid)
 	list = tcb->group->tg_socketlist;
 	sock = (struct lwip_sock *)list.sl_sockets[s].sock;
 
-	if (!sock || !sock->conn) {
+	if (!sock) {
```

Comment:
Is this enough conditional? Don't you need to compare to `s`?

Reply:
It tries to get the socket from its task_group based on 's'. So I think it is enough.